packages/devtools/lib/src/memory/memory.dart

```diff
@@ -185,6 +228,205 @@ class MemoryScreen extends Screen with SetStateMixin {
     return screenDiv;
   }
 
+  ClassHeapDetailStats _searchClass(String className) {
+    final List<ClassHeapDetailStats> classesData = tableStack.first.data;
+    for (ClassHeapDetailStats stat in classesData) {
+      if (stat.classRef.name == className) {
+        return stat;
+      }
+    }
+
+    return null;
+  }
+
+  void _selectClass(String className, [record = true]) {
+    final List<ClassHeapDetailStats> classesData = tableStack.first.data;
+    int row = 0;
+    for (ClassHeapDetailStats stat in classesData) {
+      if (stat.classRef.name == className) {
+        tableStack.first.selectByIndex(row, scrollBehavior: 'auto');
+        if (record) {
+          memoryPath.add(NavigationState.classSelect(className));
+        }
+        return;
+      }
+      row++;
+    }
+
+    framework.toast('Unable to find class $className', title: 'Error');
+  }
+
+  // Show the spinner we're computing...
+  Spinner createSpinner(Table<Object> table) {
+//    final html.Rectangle r = table.element.element.getBoundingClientRect();
+//    final int topPosition = r.top + (r.height / 2).round();
+//    final int leftPosition = r.left + (r.width / 2).round();
+
+    final Spinner spinner = Spinner.centered();
+//TODO(terry): Cleanup
+    //final Spinner spinner = Spinner()..clazz('padded');
+/*
+    spinner.element.style
+      ..position = 'absolute'
+      ..top = '${topPosition}px'
+      ..left = '${leftPosition}px'
+      ..zIndex = '1000';
+*/
+    return spinner;
+  }
+
+  Future<int> _selectInstanceInFieldHashCode(
+      String field, String hashCode) async {
+    final Table<Object> instanceTable = tableStack.elementAt(1);
+    final Spinner spinner = createSpinner(instanceTable);
+    instanceTable.element.add(spinner);
+
+    // There's an instances table up.
+    final List<InstanceSummary> instances = instanceTable.data;
+    int row = 0;
+    for (InstanceSummary instance in instances) {
+      // Check the field in each instance looking to find the object being held
+      // (the hashCode passed in matches the particular field's hashCode)
+      final evalResult = await evaluate(instance.objectRef, '$field.hashCode');
```

Comment:
Fyi. @bkonyi 
we now have a case where we are calling eval on 1000s of objects.
Granted it would be ideal if this code didn't have to use eval but that is a separate issue

Reply:
First thing I see here is that we're awaiting each evaluation which will slow things down significantly more than the overhead of individual calls to the service, so this block of code should likely be rewritten using more of the futures API to block on a batch of requests.

I haven't taken a deep look at this PR, but I'd be surprised if there wasn't a better way to accomplish this... I'll take a closer look tomorrow when I'm hopefully back in the office.